### ISSUE TITLE
go mod tidy fails against v0.6.2 using go 1.13.8

### ISSUE THREAD

**Issue by treystout** [slack-go/slack#666]
```
$ go clean -modcache
$ go mod tidy
go: downloading github.com/slack-go/slack v0.6.2
...EXTRA STUFF OMITTED...
verifying github.com/slack-go/slack@v0.6.2: checksum mismatch
        downloaded: h1:MAnjpRC+m082cFXT/BOYZOrlrZXJZIORfLqw9IwK/hI=
        go.sum:     h1:sSqeyJIQF/DhoP61BJscqJAdihctf+X+EttAGYZ44YY=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```
Package install using golang 1.12.15 doesn't have this issue. I am on amd64 platform. What else can I report to help?

**Comment by ptxmac** [comment]
I have this problem too on go version 1.13.7

**Comment by kanata2** [comment]
It seems that a checksum of go.sum(= `h1:sSqey...`) is for go 1.12.
So please remove a line which starts with `github.com/slack-go/slack v0.6.2` in go.sum file and try `go clean -modcache`, `go mod tidy` again.

**Comment by vrischmann** [comment]
Same issue here with go 1.13.7. Reverting to v0.6.1 works.

I'm pretty sure I was able to use v0.6.2 last week with go 1.13.6 and 1.13.8, not sure what happened.

**Comment by kanata2** [comment]
Could you try https://github.com/slack-go/slack/issues/666#issuecomment-593141216 ?

**Comment by vrischmann** [comment]
I grepped for that sum `h1:sSqeyJIQF/DhoP61BJscqJAdihctf+X+EttAGYZ44YY=` I can't find it anywhere.

Note that for me the mismatch is different:

```
$ go mod tidy
go: downloading github.com/slack-go/slack v0.6.2
verifying github.com/slack-go/slack@v0.6.2: checksum mismatch
        downloaded: h1:sSqeyJIQF/DhoP61BJscqJAdihctf+X+EttAGYZ44YY=
        go.sum:     h1:MAnjpRC+m082cFXT/BOYZOrlrZXJZIORfLqw9IwK/hI=
```

then if I remove the lines from my `go.sum` file I get this mismatch:

```
$ go mod tidy
go: downloading github.com/slack-go/slack v0.6.2
verifying github.com/slack-go/slack@v0.6.2: checksum mismatch
        downloaded: h1:sSqeyJIQF/DhoP61BJscqJAdihctf+X+EttAGYZ44YY=
        sum.golang.org: h1:MAnjpRC+m082cFXT/BOYZOrlrZXJZIORfLqw9IwK/hI=
```

**Comment by kanata2** [comment]
Which version did you use in https://github.com/slack-go/slack/issues/666#issuecomment-593360321?

**Comment by vrischmann** [comment]
Not sure which version you're asking about, it was Go 1.13.7 provided by Fedora.

I'm on another computer right now and I can't reproduce the problem, with neither Go 1.13.7/1.13.8/1.14 from [here](https://golang.org/dl/).

I'll try to reproduce when I get back home.

**Comment by vrischmann** [comment]
I'm back at my home computer. I was using the offical Go 1.13.7 distribution, not the Fedora one. I tried using the Fedora package (which is Go 1.13.6) and it's not working.

I again downloaded Go 1.13.7/1.13.8/1.14 from the official website and it wasn't working either.

But in the end I found the issue, I was trying to think what about my home setup is different than my work setup, and the difference is that I'm using a private GOPROXY at work and at home I have `GOPROXY=direct`.

Commenting that in my `$HOME/.config/go/env` fixes the problem, so it looks like the v0.6.2 that sumdb saw is not identical to the v0.6.2 tag that is fetched when not using a proxy. Maybe the tag was force-pushed ?

**Comment by gjgd** [comment]
I got the same error here since we updated to 0.6.2: https://github.com/gjgd/dont-slack-evil/runs/482882834#step:6:52

I tried https://github.com/slack-go/slack/issues/666#issuecomment-593141216 and it did not help

```
➜ go version
go version go1.13.8 darwin/amd64
```

**Comment by pkutishch** [comment]
Hit the same issue today, but works fine with v0.6.1. there is indeed checksum mismatch

**Comment by parsley42** [comment]
I'm not sure what's going on here; my project is using slack v0.6.2, and here are the relevant lines from "go mod tidy" using Go 1.14:
```
go: finding module for package github.com/slack-go/slack
...
go: found github.com/slack-go/slack in github.com/slack-go/slack v0.6.2
```
Then, grep'ing slack in go.*:
```
go.mod:	github.com/slack-go/slack v0.6.2
go.sum:github.com/slack-go/slack v0.6.2 h1:MAnjpRC+m082cFXT/BOYZOrlrZXJZIORfLqw9IwK/hI=
go.sum:github.com/slack-go/slack v0.6.2/go.mod h1:ZUNi+O1Pwr2ch2UOp2AfF+s7QYQgwht2Cd1UTeIYw9A=
```
Note that I haven't looked deeply in to this at all, I just wanted to provide a datapoint and a "Works For Me". I realize the Go version differs, but I don't know why that would matter.

**Comment by gjgd** [comment]
I solved the problem by using go 1.13 instead of go 1.12

**Comment by kanata2** [comment]
I'll close this issue, but feel free to re-open if you come across this again!
